common/flatpak-dir.c

```diff
@@ -293,8 +293,18 @@ flatpak_remote_state_ensure_summary (FlatpakRemoteState *self,
                                      GError            **error)
 {
   if (self->summary == NULL)
-    return flatpak_fail_error (error, FLATPAK_ERROR_INVALID_DATA, _("Unable to load summary from remote %s: %s"), self->remote_name,
-                               self->summary_fetch_error != NULL ? self->summary_fetch_error->message : "unknown error");
+    {
+      if (self->summary_fetch_error != NULL &&
```

Comment:
Rather than (or perhaps as well as?) handling `G_IO_ERROR_CANCELLED` everywhere `summary_fetch_error` and `metadata_fetch_error` are read, would it make more sense to handle it in `_flatpak_dir_get_remote_state()`, where the cancellation originally happens?

i.e. Presumably if the `GCancellable` passed in to `_flatpak_dir_get_remote_state()` is cancelled, then we might actually just want `_flatpak_dir_get_remote_state()` to immediately return `G_IO_ERROR_CANCELLED`, rather than saving that error and trying to continue in `optional` mode.

Reply:
Yeah, I agree. We also want the call to _flatpak_dir_get_remote_state() to return the same kind of errors as flatpak_remote_state_ensure_summary(), so it makes sense to do the translation there before setting summary_fetch_error.